### PR TITLE
Don't pass contacts without custom fields to the client

### PR DESCRIPTION
**Pull request by cammellos**
When receiving a message with save a contact in the database in order to
avoid re-calculating image/profile.
This contact is then passed to the client, which can negatively impact
performance.

This commit changes the behavior so that only those contacts that have
some custom fields (have been explicitly added by the user, have been
blocked by the user, have sent a contact request or have a verified ens
name) are passed to the client.

### CODE REVIEW

File: protocol/messenger_test.go [hunk 1]
```diff
@@ -1191,7 +1191,7 @@ func (s *MessengerSuite) TestBlockContact() {
 		Name:        "contact-name",
 		Photo:       "contact-photo",
 		LastUpdated: 20,
-		SystemTags:  []string{"1", "2"},
+		SystemTags:  []string{contactAdded, contactRequestReceived},
```
Comment (Samyoul) [inline protocol/messenger_test.go:1194]: I like the use of constants rather than the inline declaration. 👍

**Review by Samyoul — APPROVED** [review]
LGTM

File: protocol/messenger.go [hunk 2]
```diff
@@ -2009,8 +2021,8 @@ func (m *Messenger) handleRetrievedMessages(chatWithMessages map[transport.Filte
 		}
 	}
 
-	if len(messageState.Response.Contacts) > 0 {
-		err = m.persistence.SaveContacts(messageState.Response.Contacts)
+	if len(contactsToSave) > 0 {
+		err = m.persistence.SaveContacts(contactsToSave)
```
Comment (jakubgs) [inline protocol/messenger.go:2025]: Wouldn't using `messageState.ModifiedContacts` here make sense?

File: protocol/messenger_installations_test.go [hunk 3]
```diff
@@ -250,28 +250,6 @@ func (s *MessengerInstallationSuite) TestSyncInstallation() {
...
 	s.Require().NotNil(statusChat)
 
-	var actualContact *Contact
-	for _, c := range allContacts {
-		if c.ID == contact.ID {
-			actualContact = c
-		}
-	}
-
+	actualContact := allContacts[0]
```
Comment (jakubgs) [inline protocol/messenger_installations_test.go]: Why are we just picking the first one here? I assume it's the only one with custom fields, but could we make this more explicit?

**Review by jakubgs — APPROVED** [review]
Makes sense to me.

File: protocol/messenger.go (hunk 2 above)
Comment (cammellos) [reply protocol/messenger.go:2025]: `ModifiedContacts` contains only the ID of the one that have  been modified, we need to hydrate them (pull them from `allContacts`)

File: protocol/messenger_installations_test.go (hunk 3 above)
Comment (Samyoul) [reply protocol/messenger_installations_test.go]: @jakubgs how would you propose making this more explicit? Adding a comment? Changing the variable name?

Comment (Samyoul) [reply protocol/messenger_installations_test.go]: @cammellos is there a check that the `allContacts` slice has a len > 0?

Comment (cammellos) [reply protocol/messenger_installations_test.go]: Yes, we would throw an error if the slice has length `0` (in `RetryWithBackOff` we return an error until `len(x) == 1`)

**Pull request merged**